src/main/cpp/src/get_json_object.cu

```diff
@@ -1191,6 +933,54 @@ __device__ thrust::pair<bool, size_t> get_json_object_single(
   return {success, generator.get_output_len()};
 }
 
+/**
+ * @brief Kernel for running the JSONPath query.
+ *
+ * This kernel operates in a 2-pass way. On the first pass it computes the
+ * output sizes. On the second pass, it fills in the provided output buffers
+ * (chars and validity).
+ *
+ * @param col Device view of the incoming string
+ * @param commands JSONPath command buffer
+ * @param output_offsets Buffer used to store the string offsets for the results
+ *        of the query
+ * @param out_buf Buffer used to store the results of the query
+ * @param out_validity Output validity buffer
+ * @param out_valid_count Output count of # of valid bits
+ * @param options Options controlling behavior
+ */
+template <int block_size>
+__launch_bounds__(block_size) CUDF_KERNEL
+  void get_json_object_size_kernel(cudf::column_device_view col,
+                                   cudf::device_span<path_instruction const> path_commands,
+                                   cudf::size_type* d_sizes,
+                                   cudf::detail::input_offsetalator output_offsets)
+{
+  auto tid          = cudf::detail::grid_1d::global_thread_id();
+  auto const stride = cudf::detail::grid_1d::grid_stride();
+
```

Comment:
No need to sync.
`__ballot_sync` is used to sync to generate a 32 bits mask within 32 threads(in a warp).
This `get_json_object_size_kernel` does not touch any mask, so need to do sync.

Reply:
done.